### DISCUSSION TITLE
How to transfer raw data after contract negotiation?

### DISCUSSION THREAD

**Discussion by dilip-dhankecha** [eclipse-edc/Connector#1096] [Q&A] [answered]
I referred all samples, so far it looks good and I am able to run this in my local.

Now, let's say I have some data available in my database or in xls file and I want to share this data(Not the entire file, I want to share some specific information) with consumers as an asset, so In this case, once the contract negotiation is completed how the consumer will get the data?

**Comment by jimmarino** [comment, marked as answer]
You can use:

1. One of the supported data transfer extensions (e.g. S3, Azure Object Storage)
2. The data plane framework (HTTP or custom endpoints)
3. Your own implementation via DataFlowController

This is explained in the samples and onboarding tutorials.
